src/torch_ucc_comm.cpp

```diff
@@ -133,23 +139,21 @@ ucc_status_t oob_allgather_test(void* req) {
 }
 
 ucc_status_t oob_allgather_free(void* req) {
-  torch_ucc_oob_coll_info_t* info =
-      reinterpret_cast<torch_ucc_oob_coll_info_t*>(req);
+  auto* info = reinterpret_cast<torch_ucc_oob_coll_info_t*>(req);
   try {
-    int num_done = info->store->add({info->getKey("ag_done")}, 1);
+    int num_done = info->store->add({info->getKey(kAllGatherDone)}, 1);
     if (num_done == info->size) {
-      info->store->deleteKey(info->getKey("ag_done"));
-      for (int r = 0; r < info->size; r++) {
-        info->store->deleteKey(info->getKey("teamr" + std::to_string(r)));
-      }
-      for (int r = 0; r < info->size; r++) {
-        info->store->add({info->getKey("ag_free" + std::to_string(r))}, 1);
+      info->store->deleteKey(info->getKey(kAllGatherDone));
+      for (const auto r : c10::irange(info->size)) {
+        info->store->deleteKey(info->getKey(kTeamR + std::to_string(r)));
+        info->store->add({info->getKey(kAllGatherFree + std::to_string(r))}, 1);
```

Comment:
@Sergei-Lebedev - we're combing `deleteKey` and `add` here, do you recall are there any specific reasons we use two loops here?

(cc @vtlam)

Reply:
Thanks. I will add a comment since this is critical.